exporting/exporting_engine.c

```diff
@@ -4,6 +4,55 @@
 
 static struct engine *engine = NULL;
 
+/**
+ * Exporting Clean Engine
+ *
+ * Clean all variables allocated inside engine structure
+ *
+ * @param en a pointer to the strcuture that will be cleaned.
+ */
+static void exporting_clean_engine()
+{
+    if (!engine)
+        return;
+
+    int disable_awk_sdk = 0;
+#if ENABLE_PROMETHEUS_REMOTE_WRITE
+    int disable_remote_write = 0;
+#endif
+    for (struct instance *instance = engine->instance_root; instance;) {
+        struct instance *current_instance = instance;
+        instance = instance->next;
+
+        if (current_instance->config.type == EXPORTING_CONNECTOR_TYPE_PROMETHEUS_REMOTE_WRITE) {
+#if ENABLE_PROMETHEUS_REMOTE_WRITE
+            clean_prometheus_remote_write_instance(instance);
```

Comment:
You deleted it, but we still need it. It's a specific cleanup function, we can't leave it for the default simple connector worker cleanup.

Reply:
I brought back the prometheus cleanup, but we won't clean `worker_path`, because we receive `double free corruption`. This is happening, because the compiler is doing optimizations that eliminates the necessity to clean it.